src/overlay/OverlayManagerImpl.cpp

```diff
@@ -391,6 +426,34 @@ OverlayManagerImpl::connectTo(std::vector<PeerBareAddress> const& peers,
     return count;
 }
 
+void
+OverlayManagerImpl::maybeRefreshPeerIPs()
+{
+    CLOG(TRACE, "Overlay")
+        << "Checking IPs for PREFERRED_PEERS and KNOWN_PEERS";
+
+    try
+    {
+        for (auto const& s : mApp.getConfig().PREFERRED_PEERS)
+        {
+            getLatestPeerAddress(s, true);
+        }
+        for (auto const& s : mApp.getConfig().KNOWN_PEERS)
+        {
+            getLatestPeerAddress(s, false);
+        }
+    }
+    catch (std::runtime_error& e)
+    {
+        CLOG(ERROR, "Overlay")
```

Comment:
This fails after just the first error and does not let other peers get resolved even if the error only occurs on first one. Moving this try/catch inside getLatestPeerAddress would fix that issue.